jellyfin-core/src/main/kotlin/org/jellyfin/sdk/discovery/RecommendedServerDiscovery.kt

```diff
@@ -78,29 +111,31 @@ public class RecommendedServerDiscovery(
 		)
 		val api = SystemApi(client)
 
-		val info: Response<PublicSystemInfo>?
+		val info: Result<Response<PublicSystemInfo>>
 		val responseTime = measureTimeMillis {
 			@Suppress("TooGenericExceptionCaught")
 			info = try {
-				api.getPublicSystemInfo()
+				val response = api.getPublicSystemInfo()
+				if (response.status == HTTP_OK) Result.success(response)
+				else Result.failure(InvalidStatusException(response.status))
 			} catch (err: TimeoutException) {
 				logger.debug("Could not connect to $address", err)
-				null
+				Result.failure(err)
 			} catch (err: InvalidStatusException) {
 				logger.debug("Received unexpected status ${err.status} from $address", err)
-				null
+				Result.failure(err)
 			} catch (err: InvalidContentException) {
 				logger.debug("Could not parse response from $address", err)
-				null
+				Result.failure(err)
 			} catch (err: Throwable) {
 				logger.error("Could not retrieve public system info for $address", err)
-				null
+				Result.failure(err)
 			}
 		}
 
 		return SystemInfoResult(
 			address = address,
-			systemInfo = if (info != null && info.status == HTTP_OK) info.content else null,
+			systemInfo = info.map { it.content },
```

Comment:
I'd recommend using a reference lambda here to eliminate the `it`.

Reply:
I changed this line